src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -478,7 +478,30 @@ public static IEnumerator GetEnumerator(IsolatedStorageScope scope)
             // (say, for a particular StrongName). You could iterate any store opened at least once by NetFX on
             // NetFX as it would create the needed identity file. You wouldn't be able to iterate if it was only
             // ever opened by CoreFX, as the needed file isn't there yet.
-            throw new PlatformNotSupportedException();
+            return new IsolatedStorageFileEnumerator();
```

Comment:
I appreciate and understand the feedback. One of the things I wasn't clear on is that I did this to make it easier to compare the code between CoreFX/NetFX. I agree it is incredibly unlikely that someone will check/reflect the type, but these sort of things do happen, and as it was trivial and had the benefit of being a little easier to compare code I went the way I did.

While `Array.Empty<IsolatedStorageFile>().GetEnumerator()` has effectively the same behavior (exceptions included), it requires an extra level of mental digging ("are these really effectively similar?") when comparing the two codebases. Given the high level of complexity of the NetFX (and variants with Silverlight) code I'm consciously trying to keep things structurally similar- unless I can significantly reduce the complexity (and in those cases I'm making an effort leave detailed comments). I expect we'll have to keep coming back to this code over the next several years- hopefully I've done enough to make that as easy as possible.